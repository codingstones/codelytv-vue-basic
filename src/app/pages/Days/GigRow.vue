<template>
  <q-card @click.native="onClick(gig)">
    <q-item>
      <q-item-side :image="gig.image_url" />
      <q-item-main>
        <q-item-tile label>{{gig.title}}</q-item-tile>
        <q-item-tile sublabel>{{gig.place}}</q-item-tile>
      </q-item-main>
    </q-item>
  </q-card>
</template>

<script>
  export default {
    props: { gig: Object, onClick: Function }
  }
</script>

<style lang="stylus" type="text/stylus">
  .q-item:hover {
    background: alpha(#795548, 30%);
    cursor: pointer;
  }
</style>
