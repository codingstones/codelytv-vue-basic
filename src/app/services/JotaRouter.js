import VueRouter from 'vue-router'

export const JotaRouter = (router) => {
  if (!router) router = new VueRouter()
  return { navigateToGig, navigateToCreateGig, navigateToAllGigs, getParam }

  function navigateToGig(gigId) {
    console.log('ENTRO POR AQUI CO')
    router.push('gig/' + gigId)
  }

  function navigateToCreateGig() {
    router.push({path: '/newGig'})
  }

  function navigateToAllGigs() {
    router.push({path: '/all'})
  }

  function getParam(id) {
    if (!router) return ''
    return router.currentRoute.params[id]
  }
}

export const jotaRouterMixin = {
  created: function () {
    this.jotaRouter = JotaRouter(this.$router)
  }
}
