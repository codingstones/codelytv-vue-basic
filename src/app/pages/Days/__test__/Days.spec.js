import { mount } from '@vue/test-utils'
import Days from '@/app/pages/Days/Days.vue'
import { FIRST_DAY, DAY_LIST } from '../../../services/__mocks__/gigs-sample'
import DayListPage from '../../../__page_objects__/DaysPageObject'
import { localizedFromIso } from '../../../services/date-utils'
jest.mock('@/app/services/jota-api')

describe('Days', () => {
  const FIRST_DAY_GIG_TITLES = FIRST_DAY.gigs.map(gig => gig.title)

  let page, wrapper
  beforeEach(async () => {
    wrapper = mount(Days)
    page = new DayListPage(wrapper)
  })

  it('matches full snapshot', async() => {
    page.matchSnapshot()
  })

  it('renders all gigs in the first day', async() => {
    FIRST_DAY_GIG_TITLES.map((text) => page.contains(text))
  })

  describe('When clicking buttons', () => {

    let navigateToGigSpy
    beforeEach(() => {
      navigateToGigSpy = jest.fn()
      wrapper.vm.jotaRouter.navigateToGig = navigateToGigSpy
    })

    it('navigates to first gig detail', async () => {
      const FIRST_GIG = FIRST_DAY.gigs[0]
      page.clickFirstGig()
      expect(navigateToGigSpy).toHaveBeenCalledWith(FIRST_GIG.id)
    })

    it('navigates to second gig detail', async () => {
      const SECOND_GIG = FIRST_DAY.gigs[1]
      page.clickSecondGig()
      expect(navigateToGigSpy).toHaveBeenCalledWith(SECOND_GIG.id)
    })
  })

  /* Different examples of more accurate tests that need
  to explicitly run over the DOM structure
  */

  it('render days in localized format', async () => {
    const DAYS = DAY_LIST.map((day) => localizedFromIso(day.date))
    expect(page.dayTitles()).toEqual(DAYS)
  })

  it('render gigs for each day', async () => {
    let expectedGigs
    DAY_LIST.map((day, index) => {
      expectedGigs = day.gigs.map((gig) => gig.title + ' ' + gig.place)
      expect(page.gigRowsFor(index)).toEqual(expectedGigs)
    })
  })
})
