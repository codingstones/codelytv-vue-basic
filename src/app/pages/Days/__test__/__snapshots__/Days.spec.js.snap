// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Days matches full snapshot 1`] = `
<div class="full-width">
  <!---->
  <div>
    <div>
      <div class="day">
        <div class="q-card-primary q-card-container row no-wrap">
          <div class="col column">
            <div class="q-card-title">Lunes, 18 De Septiembre</div>
            <div class="q-card-subtitle"></div>
          </div>
          <div class="col-auto self-center q-card-title-extra"></div>
        </div>
        <div>
          <div class="q-card">
            <div class="q-item q-item-division relative-position">
              <div class="q-item-side q-item-section q-item-side-left"><img src="http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png" class="q-item-image"></div>
              <div class="q-item-main q-item-section">
                <div class="q-item-label">Anarchy in the Jota S</div>
                <div class="q-item-sublabel">SCBCN</div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div class="q-card">
            <div class="q-item q-item-division relative-position">
              <div class="q-item-side q-item-section q-item-side-left"><img src="http://www.wtf.es/wtf.png" class="q-item-image"></div>
              <div class="q-item-main q-item-section">
                <div class="q-item-label">Jotas de Codely TV</div>
                <div class="q-item-sublabel">Around the world</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div> <button tabindex="0" type="button" class="q-btn inline relative-position q-btn-item non-selectable fixed-bottom-right q-btn-round q-focusable q-hoverable bg-secondary text-white hidden" style="margin: 0px 15px 15px 0px;">
    <div class="q-focus-helper"></div>
    <div class="q-btn-inner row col items-center justify-center"><i aria-hidden="true" class="q-icon material-icons">keyboard_arrow_up</i></div>
  </button>
</div>
`;
